src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioDocumentNavigationService.cs

```diff
@@ -357,15 +370,17 @@ private bool TryNavigateToMappedFile(Workspace workspace, Document generatedDocu
             return false;
         }
 
-        private MappedSpanResult? GetMappedSpan(ISpanMappingService spanMappingService, Document generatedDocument, TextSpan textSpan, CancellationToken cancellationToken)
+        private static async Task<MappedSpanResult?> GetMappedSpanAsync(
+            ISpanMappingService spanMappingService, Document generatedDocument, TextSpan textSpan, CancellationToken cancellationToken)
         {
             // Mappings for opened razor files are retrieved via the LSP client making a request to the razor server.
             // If we wait for the result on the UI thread, we will hit a bug in the LSP client that brings us to a code path
             // using ConfigureAwait(true).  This deadlocks as it then attempts to return to the UI thread which is already blocked by us.
             // Instead, we invoke this in JTF run which will mitigate deadlocks when the ConfigureAwait(true)
             // tries to switch back to the main thread in the LSP client.
             // Link to LSP client bug for ConfigureAwait(true) - https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1216657
```

Comment:
Is this comment applicable anymore?

Reply:
oh that's terrifying.

Reply:
this should no longer apply as we're either all async, or our callers need to be in JTF run calls themselves (this PR).